python/lsst/meas/algorithms/sourceSelector.py

```diff
@@ -97,3 +103,301 @@ def _isBad(self, source):
+
+        Returns whether the source is selected.
+
+        Parameters
+        ----------
+        catalog : `lsst.afw.table.SourceCatalog`
+            Source to which to apply limits.
+
+        Returns
+        -------
+        selected : `numpy.ndarray`
+            Boolean array indicating whether each source is selected.
+        """
+        selected = np.ones(len(catalog), dtype=bool)
+        for flag in self.good:
+            selected &= catalog[flag]
+        for flag in self.bad:
+            selected &= ~catalog[flag]
+        return selected
+
+
+class RequireUnresolved(pexConfig.Config):
+    """Select sources using star/galaxy separation
+
+    This object can be used as a `lsst.pex.config.Config` for configuring
+    the limit, and then the `apply` method can be used to test sources in
+    a catalog against the configured limit.
+    """
+    name = pexConfig.Field(dtype=str, default="base_ClassificationExtendedness_value",
+                           doc="Name of column for star/galaxy separation")
+    minimum = pexConfig.Field(dtype=float, optional=True, doc="Select objects with value greater than this.")
+    maximum = pexConfig.Field(dtype=float, optional=True, default=0.5,
+                              doc="Select objects with value less than this.")
+
+    def apply(self, catalog):
+        """Apply the flag requirements to a catalog
+
+        Returns whether the source is selected.
+
+        Parameters
+        ----------
+        catalog : `lsst.afw.table.SourceCatalog`
+            Source to which to apply limits.
+
+        Returns
+        -------
+        selected : `numpy.ndarray`
+            Boolean array indicating whether each source is selected.
+        """
+        selected = np.ones(len(catalog), dtype=bool)
+        value = catalog[self.name]
+        if self.minimum is not None:
+            selected &= value > self.minimum
+        if self.maximum is not None:
+            selected &= value < self.maximum
+        return selected
+
+
+class ScienceSourceSelectorConfig(pexConfig.Config):
+    """Configuration for selecting science sources"""
+    doFluxLimit = pexConfig.Field(dtype=bool, default=False, doc="Apply flux limit?")
+    doFlags = pexConfig.Field(dtype=bool, default=False, doc="Apply flag limitation?")
+    doUnresolved = pexConfig.Field(dtype=bool, default=False, doc="Apply unresolved limitation?")
+    fluxLimit = pexConfig.ConfigField(dtype=FluxLimit, doc="Flux limit to apply")
+    flags = pexConfig.ConfigField(dtype=RequireFlags, doc="Flags to require")
+    unresolved = pexConfig.ConfigField(dtype=RequireUnresolved, doc="Star/galaxy separation to apply")
+
+    def setDefaults(self):
+        pexConfig.Config.setDefaults(self)
+        self.flags.bad = ["base_PixelFlags_flag_edge", "base_PixelFlags_flag_interpolated",
+                          "base_PixelFlags_flag_saturated"]
+
+class ScienceSourceSelectorTask(BaseSourceSelectorTask):
+    """Science source selector
```

Comment:
What does "science" mean in this context? e.g. I'd worry about using a strict magnitude or star/galaxy cut for defining "science worthy" sources. I'm not sure I have a better name for it, though.

Reply:
Added an explanation of "science":

>    By "science" sources, we mean sources that are on images that we
>    are processing, as opposed to sources from reference catalogs.